components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -117,11 +117,14 @@ func (p *pg) PurgeSubjectFromPolicies(ctx context.Context, sub string) ([]string
 	// prescribe this, but it feels like the better choice.)
 
 	row := tx.QueryRowContext(ctx, `
-WITH pol_ids AS (DELETE FROM iam_policy_members
-                 WHERE member_id=(SELECT id FROM iam_members WHERE name=$1)
-                 RETURNING policy_id)
-SELECT array_agg(policy_id) FROM pol_ids`,
-		sub)
+		WITH pol_db_ids AS (
+			DELETE FROM iam_policy_members
+			WHERE member_id=(SELECT db_id FROM iam_members WHERE name=$1)
+			RETURNING policy_id
+		)
+		SELECT array_agg(id)
+		FROM iam_policies
+		WHERE db_id IN (SELECT * FROM pol_db_ids);`, sub)
```

Comment:
Hmm I guess we could also do this

```sql
SELECT array_agg(id)
FROM iam_policies
WHERE db_id IN (
  DELETE FROM iam_policy_members
  WHERE member_id=(
    SELECT db_id
    FROM iam_members
    WHERE name=$1)
  RETURNING policy_id)
```
Hmmm I think I like yours better.

Reply:
it's a toss up, but I like that the `DELETE` leads the first one (which seems like the right one for a method with the word `Purge` in it)